### ISSUE TITLE
Multiplying transposed BoolTensor with FloatTensor

### ISSUE THREAD

**Issue by gabrielhuang** [pytorch/pytorch#31174]
## To Reproduce

```python
import torch

print('Version', torch.__version__)
# 1.3.0 for me

f = torch.arange(4*4, dtype=torch.float).view(4, 4)
print('f\n', f)
# f
#  tensor([[ 0.,  1.,  2.,  3.],
#         [ 4.,  5.,  6.,  7.],
#         [ 8.,  9., 10., 11.],
#         [12., 13., 14., 15.]])
g = torch.zeros((4, 4), dtype=torch.bool)
g[:, :1] = True
print('g\n', g)
# g
#  tensor([[ True, False, False, False],
#         [ True, False, False, False],
#         [ True, False, False, False],
#         [ True, False, False, False]])
print('Wrong: f*g.t()\n', f * (g.t()))
# Wrong: f*g.t()
#  tensor([[ 0.,  0.,  0.,  0.],
#         [ 4.,  0.,  0.,  0.],
#         [ 8.,  0.,  0.,  0.],
#         [12.,  0.,  0.,  0.]])
print('Correct: f*g.t().clone()\n', f * (g.t().clone()))
# Correct: f*g.t().float()
#  tensor([[0., 1., 2., 3.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.]])
print('Correct: f*g.t().float()\n', f * (g.t().float()))
# Correct: f*g.t().float()
#  tensor([[0., 1., 2., 3.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.]])
print('Correct: f*g.float().t() \n', f * (g.float().t()))
# Correct: f*g.float().t()
#  tensor([[0., 1., 2., 3.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.],
#         [0., 0., 0., 0.]])
```

## Expected behavior

The "wrong" case should have matched the "correct" cases.

## Environment

```
PyTorch version: 1.3.0
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Debian GNU/Linux rodete
GCC version: (Debian 8.3.0-6) 8.3.0
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Quadro P1000
Nvidia driver version: 430.50
cuDNN version: Could not collect

Versions of relevant libraries:
[pip3] numpy==1.17.2
[pip3] torch==1.3.0
[conda] Could not collect
```

## Additional context

I am using those masks in the context of Transformers.
Since it does not raise errors, I naturally tend to not convert BoolTensor masks to float before multiplying them with FloatTensors.
It seems like casting to .float() solves the problem.

**Comment by colesbury** [comment]
Sorry you ran into this. This is fixed in 1.3.1. Here are the release notes for reference: https://github.com/pytorch/pytorch/releases

**Comment by gabrielhuang** [comment]
Good to know it's already been fixed. Thank you !
